File: backend/src/modules/pdt/pdt.controller.ts
import { Controller, Get, Post, Patch, Delete, Body, Query, Param, ParseIntPipe } from '@nestjs/common';
import { ApiTags, ApiOperation } from '@nestjs/swagger';
import { PdtService } from './pdt.service';
import { PaginationDto } from '../../common/dto/pagination.dto';
import { ApiResponseDto } from '../../common/dto/api-response.dto';

@ApiTags('前缀域转换')
@Controller('api/pdt')
export class PdtController {
  constructor(private readonly pdtService: PdtService) {}

  @Get()
  @ApiOperation({ summary: '获取 PDT 列表' })
  async findAll(@Query() dto: PaginationDto) {
    return ApiResponseDto.success(await this.pdtService.findAll(dto));
  }

  @Post()
  async create(@Body() data: any) {
    return ApiResponseDto.success(await this.pdtService.create(data), '创建成功');
  }

  @Patch(':id')
  async update(@Param('id', ParseIntPipe) id: number, @Body() data: any) {
    return ApiResponseDto.success(await this.pdtService.update(id, data), '更新成功');
  }

  @Delete(':id')
  async remove(@Param('id', ParseIntPipe) id: number) {
    await this.pdtService.remove(id);
    return ApiResponseDto.success(null, '删除成功');
  }

  @Post('reload')
  @ApiOperation({ summary: '重载 PDT' })
  async reload() {
    await this.pdtService.reload();
    return ApiResponseDto.success(null, '重载成功');
  }

  @Get('list')
  @ApiOperation({ summary: '获取内存列表' })
  async list() {
    return ApiResponseDto.success(await this.pdtService.list());
  }
}

File: kam-admin-module-routing/src/main/java/io/kamailio/admin/modules/pdt/controller/PdtController.java
package io.kamailio.admin.modules.pdt.controller;

import io.kamailio.admin.common.dto.ApiResponse;
import io.kamailio.admin.common.dto.PaginatedResult;
import io.kamailio.admin.common.dto.PaginationDto;
import io.kamailio.admin.modules.pdt.entity.Pdt;
import io.kamailio.admin.modules.pdt.service.PdtService;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.tags.Tag;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@Tag(name = "前缀域转换")
@RestController
@RequestMapping("/api/pdt")
public class PdtController {
    private final PdtService service;

    public PdtController(PdtService service) { this.service = service; }

    @GetMapping
    @Operation(summary = "获取 PDT 列表")
    public ResponseEntity<ApiResponse<PaginatedResult<Pdt>>> findAll(PaginationDto dto) {
        return ResponseEntity.ok(ApiResponse.success(service.findAll(dto)));
    }

    @PostMapping
    public ResponseEntity<ApiResponse<Pdt>> create(@RequestBody Pdt data) {
        return ResponseEntity.ok(ApiResponse.success(service.create(data), "创建成功"));
    }

    @PatchMapping("/{id}")
    public ResponseEntity<ApiResponse<Pdt>> update(@PathVariable Integer id, @RequestBody Pdt data) {
        return ResponseEntity.ok(ApiResponse.success(service.update(id, data), "更新成功"));
    }

    @DeleteMapping("/{id}")
    public ResponseEntity<ApiResponse<Void>> remove(@PathVariable Integer id) {
        service.remove(id);
        return ResponseEntity.ok(ApiResponse.success(null, "删除成功"));
    }

    @PostMapping("/reload")
    @Operation(summary = "重载 PDT")
    public ResponseEntity<ApiResponse<Void>> reload() {
        service.reload();
        return ResponseEntity.ok(ApiResponse.success(null, "重载成功"));
    }

    @GetMapping("/list")
    @Operation(summary = "获取内存列表")
    public ResponseEntity<ApiResponse<Object>> list() {
        return ResponseEntity.ok(ApiResponse.success(service.list()));
    }
}
